compiler/luci-interpreter/include/luci_interpreter/core/Tensor.h

```diff
@@ -107,15 +107,27 @@ class Tensor
     return _quantization.zero_point[0];
   }
 
+  void allocate();
+  void deallocate();
+
   const std::vector<float> &scales() const { return _quantization.scale; }
 
   const std::vector<int32_t> &zero_points() const { return _quantization.zero_point; }
 
   int32_t quantized_dimension() const { return _quantization.quantized_dimension; }
 
-  template <typename T> const T *data() const { return reinterpret_cast<const T *>(_data.get()); }
+  template <typename T> const T *data() const
+  {
+    assert(!!_data);
```

Comment:
Just a nitpick: do you need explicit conversion to bool?

Reply:
Yes, I like this two-char idiom.